### ISSUE TITLE
console output fails on FreeBSD 10.2

### ISSUE THREAD

**Issue by teverett** [simh/simh#290]
I'm a contributor to this project (https://github.com/DoctorWkt/pdp7-unix) which uses simh. When I run pdp7 unix on OS X, I get the proper output; a login.   However on FreeBSD 10.2, it appears to hang at that point.  Here is the output from FreeBSD 10.2:

```
pdp7 unixv0.simh

PDP-7 simulator V4.0-0 Beta        git commit id: e153b7f2
CPU	idle disabled
	8192W, EAE
RB: buffering file in memory
PDP-7 simulator configuration

CPU	idle disabled
CLK	60Hz, devno=00
PTR	devno=01
PTP	devno=02
TTI	devno=03
TTO	devno=04
LPT	devno=65-66
DRM	disabled
RB	devno=71
DT	devno=75-76, 8 units
```

The expected output, from OS X is:

```
pdp7 unixv0.simh

PDP-7 simulator V4.0-0 Beta        git commit id: e153b7f2
CPU	idle disabled
	8192W, EAE
RB: buffering file in memory
PDP-7 simulator configuration

CPU	idle disabled
CLK	60Hz, devno=00
PTR	devno=01
PTP	devno=02
TTI	devno=03
TTO	devno=04
LPT	devno=65-66
DRM	disabled
RB	devno=71
DT	devno=75-76, 8 units

login: 
```

**Comment by markpizz** [comment]
Now that the #291 problem is actually fixed, I agree that this problem is not tightly related to the other one, but just for grins, can you confirm that you experience the same problem (lack of console port output) if you run the simulator directly rather than from within make?

**Comment by teverett** [comment]
I can, but it will have to be this evening.


On Wed, Mar 23, 2016 at 9:23 AM, Mark Pizzolato <notifications@github.com>
wrote:

> Now that the #291 <https://github.com/simh/simh/issues/291> problem is
> actually fixed, I agree that this problem is not tightly related to the
> other one, but just for grins, can you confirm that you experience the same
> problem (lack of console port output) if you run the simulator directly
> rather than from within make?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/simh/simh/issues/290#issuecomment-200391819>
>



-- 
A better world shall emerge based on faith and understanding  - Douglas
MacArthur

**Comment by markpizz** [comment]
Maybe today??

Since I can't reproduce this, I'll have to help you debug it,

What happens if you add the following lines to your configuration file:

    set console telnet=buffered
    set console telnet=1234

Run your test: "make run" and then from another window: telnet localhost 1234

If you get the login prompt then something about running an interactive program under make isn't working as expected.  This isn't 100% surprising to me.

If you don't get the login prompt, then other explorations will be needed.

**Comment by teverett** [comment]
I have updated to the latest simh code from github, and the latest pdp7-unix.  Here is the result of running  pdp7 directly (not from Makefile).  Notice there is no output asking for a login.

<pre>
[tom@bernice /tank/home/tom/projects/pdp7-unix/build]$ pdp7 unixv0.simh 

PDP-7 simulator V4.0-0 Beta        git commit id: d2269f8b
CPU	idle disabled
	8192W, EAE
RB: buffering file in memory
PDP-7 simulator configuration

CPU	idle disabled
CLK	60Hz, devno=00
PTR	devno=01
PTP	devno=02
TTI	devno=03
TTO	devno=04
LPT	disabled
DRM	disabled
RB	devno=71
DT	disabled



Simulation stopped, PC: 00021 (CAL 0)
sim> quit
Goodbye
RB: writing buffer to file
[tom@bernice /tank/home/tom/projects/pdp7-unix/build]$ 
</pre>

Here is the updated config files as per your instructions:

<pre>
set cpu 8k
set cpu eae
set cpu history=100
show cpu

# set up SIMH devices:

# UNIX character translations (CR to NL, ESC to ALTMODE):
set tti unix

# RB09 fixed head disk:
set rb ena
att rb image.fs

# uncomment to TELNET in GRAPHICS-2 keyboard/display(!!)
# (requires github.com/philbudne/simh)
#set g2in ena
#att -U g2in 12345

# disable hardware UNIX-7 doesn't know about:
set lpt disa
set drm disa
set dt disa

set console telnet=buffered
set console telnet=1234

# show device settings:
show dev

# load and run the paper tape bootstrap
# (loads system from disk)
load boot.rim 010000
go
</pre>

<pre>
[tom@bernice /tank/home/tom/projects/pdp7-unix/build]$ pdp7  unixv0.simh 

PDP-7 simulator V4.0-0 Beta        git commit id: d2269f8b
CPU	idle disabled
	8192W, EAE
RB: buffering file in memory
Listening on port 1234
PDP-7 simulator configuration

CPU	idle disabled
CLK	60Hz, devno=00
PTR	devno=01
PTP	devno=02
TTI	devno=03
TTO	devno=04
LPT	disabled
DRM	disabled
RB	devno=71
DT	disabled
Running with Buffered Console
</pre>

<pre>
[tom@bernice /tank/home/tom]$ telnet localhost 1234
Trying 127.0.0.1...
telnet: connect to address 127.0.0.1: Connection refused
Trying ::1...
telnet: connect to address ::1: Connection refused
telnet: Unable to connect to remote host
[tom@bernice /tank/home/tom]$ telnet localhost 1234
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.


Connected to the PDP-7 simulator CON-TEL device

</pre>

**Comment by markpizz** [comment]
OK.  We need to see if the pdp7 is actually touching the console port on your system.  I've instrumented the console output port in the attached pdp18b_stddev.c.  
1) Replace the file in the PDP18B/ directory with this one and rebuild the simulator.
2) Confirm that nothing has changed in your results
3) remove the previous changes to the configuration file
4) Add these lines to your configuration file:

      set tto debug=TRC
      set debug stdout
5) Run your simulator with the revised configuration file.

On my FreeBSD test system the output produced is:

    DBG(3)> TTO TRC: tto_reset()
    DBG(16025)> TTO TRC: tto(dev=4, pulse=1, dat=262143)
    DBG(1539840)> TTO TRC: tto(dev=4, pulse=6, dat=10)
    DBG(1540840)> TTO TRC: tto_svc(buf=012)
    DBG(1540840)> TTO TRC: tto_svc(c=012)
    
    DBG(1540852)> TTO TRC: tto(dev=4, pulse=1, dat=1)
    DBG(1540853)> TTO TRC: tto(dev=4, pulse=2, dat=1)
    DBG(1540861)> TTO TRC: tto(dev=4, pulse=6, dat=13)
    DBG(1541861)> TTO TRC: tto_svc(buf=015)
    DBG(1541861)> TTO TRC: tto_svc(c=015)
    DBG(1541873)> TTO TRC: tto(dev=4, pulse=1, dat=1)
    DBG(1541874)> TTO TRC: tto(dev=4, pulse=2, dat=1)
    DBG(12327965)> TTO TRC: tto(dev=4, pulse=6, dat=108)
    DBG(12328965)> TTO TRC: tto_svc(buf=0154)
    DBG(12328965)> TTO TRC: tto_svc(c=0154)
    lDBG(12328977)> TTO TRC: tto(dev=4, pulse=1, dat=0)
    DBG(12328978)> TTO TRC: tto(dev=4, pulse=2, dat=0)
    DBG(12329033)> TTO TRC: tto(dev=4, pulse=6, dat=111)
    DBG(12330033)> TTO TRC: tto_svc(buf=0157)
    DBG(12330033)> TTO TRC: tto_svc(c=0157)
    oDBG(12330045)> TTO TRC: tto(dev=4, pulse=1, dat=0)
    DBG(12330046)> TTO TRC: tto(dev=4, pulse=2, dat=0)
    DBG(12330101)> TTO TRC: tto(dev=4, pulse=6, dat=103)
    DBG(12331101)> TTO TRC: tto_svc(buf=0147)
    DBG(12331101)> TTO TRC: tto_svc(c=0147)
    gDBG(12331113)> TTO TRC: tto(dev=4, pulse=1, dat=0)
    DBG(12331114)> TTO TRC: tto(dev=4, pulse=2, dat=0)
    DBG(12331169)> TTO TRC: tto(dev=4, pulse=6, dat=105)
    DBG(12332169)> TTO TRC: tto_svc(buf=0151)
    DBG(12332169)> TTO TRC: tto_svc(c=0151)
    iDBG(12332181)> TTO TRC: tto(dev=4, pulse=1, dat=0)
    DBG(12332182)> TTO TRC: tto(dev=4, pulse=2, dat=0)
    DBG(12332237)> TTO TRC: tto(dev=4, pulse=6, dat=110)
    DBG(12333237)> TTO TRC: tto_svc(buf=0156)
    DBG(12333237)> TTO TRC: tto_svc(c=0156)
    nDBG(12333249)> TTO TRC: tto(dev=4, pulse=1, dat=0)
    DBG(12333250)> TTO TRC: tto(dev=4, pulse=2, dat=0)
    DBG(12333305)> TTO TRC: tto(dev=4, pulse=6, dat=58)
    DBG(12334305)> TTO TRC: tto_svc(buf=072)
    DBG(12334305)> TTO TRC: tto_svc(c=072)
    :DBG(12334317)> TTO TRC: tto(dev=4, pulse=1, dat=0)
    DBG(12334318)> TTO TRC: tto(dev=4, pulse=2, dat=0)
    DBG(12334372)> TTO TRC: tto(dev=4, pulse=6, dat=32)
    DBG(12335372)> TTO TRC: tto_svc(buf=040)
    DBG(12335372)> TTO TRC: tto_svc(c=040)
     DBG(12335384)> TTO TRC: tto(dev=4, pulse=1, dat=0)
    DBG(12335385)> TTO TRC: tto(dev=4, pulse=2, dat=0)

The numbers within the DBG(nnn) are the number of instructions the simulator has executed.  I'd expect these to be different on your system based on timing.  Notice, however, there are some lines which have the DBG( indented by one character.  Look through them and you'll see "login: ".  What is produced on your system?
[pdp18b_stddev.c.txt](https://github.com/simh/simh/files/189889/pdp18b_stddev.c.txt)

**Comment by teverett** [comment]
With the new stddev.c and the previous version of the simh file, the results are the same.  I also tried the simh file from pdp7-unix TRUNK, and the results are the same.

With your new changes to the simh here is the output

<pre>
[tom@bernice /tank/home/tom/projects/pdp7-unix/build]$ pdp7  unixv0.simh 

PDP-7 simulator V4.0-0 Beta        git commit id: d2269f8b
CPU	idle disabled
	8192W, EAE
RB: buffering file in memory
Debug output to "STDOUT"
Debug output to "STDOUT" at Fri Mar 25 12:49:36 2016
PDP-7 simulator V4.0-0 Beta        git commit id: d2269f8b
PDP-7 simulator configuration

CPU	idle disabled
CLK	60Hz, devno=00
PTR	devno=01
PTP	devno=02
TTI	devno=03
TTO	devno=04
LPT	disabled
DRM	disabled
RB	devno=71
DT	disabled

DBG(4260318795)> TTO TRC: tto_iors(INT=0)

Simulation stopped, PC: 00021 (CAL 0)
sim> quit
Goodbye
RB: writing buffer to file
Debug output disabled
[tom@bernice /tank/home/tom/projects/pdp7-unix/build]$ 
</pre>

Since the output had not changed, I checked that I had actually copied in the .c file and was running the right pdp7 binary.  

<pre>
[tom@bernice /tank/home/tom/projects/pdp7-unix/build]$ ls -lat /usr/local/bin/pdp7 
-rwxr-xr-x  1 root  wheel  543679 Mar 25 12:45 /usr/local/bin/pdp7
[tom@bernice /tank/home/tom/projects/pdp7-unix/build]$ ls -lat ~tom/projects/simh/PDP18B        
total 537
drwxr-xr-x  41 tom  wheel     84 Mar 25 12:45 ..
drwxr-xr-x   2 tom  wheel     17 Mar 25 12:44 .
-rw-r--r--   1 tom  wheel  46207 Mar 25 12:41 pdp18b_stddev.c
-rw-r--r--   1 tom  wheel  48445 Mar 20 12:04 pdp18b_sys.c
-rw-r--r--   1 tom  wheel  15588 Mar 20 12:04 pdp18b_tt1.c
-rw-r--r--   1 tom  wheel  21725 Mar 20 12:04 pdp18b_mt.c
-rw-r--r--   1 tom  wheel  11897 Mar 20 12:04 pdp18b_rb.c
-rw-r--r--   1 tom  wheel  15390 Mar 20 12:04 pdp18b_rf.c
-rw-r--r--   1 tom  wheel  22290 Mar 20 12:04 pdp18b_rp.c
-rw-r--r--   1 tom  wheel  44610 Mar 20 12:04 pdp18b_stddev.c.old
-rw-r--r--   1 tom  wheel   9731 Mar 20 12:04 pdp18b_drm.c
-rw-r--r--   1 tom  wheel  68770 Mar 20 12:04 pdp18b_dt.c
-rw-r--r--   1 tom  wheel  37135 Mar 20 12:04 pdp18b_fpp.c
-rw-r--r--   1 tom  wheel  31232 Mar 20 12:04 pdp18b_lp.c
-rw-r--r--   1 tom  wheel  99828 Mar 20 12:04 pdp18b_cpu.c
-rw-r--r--   1 tom  wheel  24386 Mar 20 12:04 pdp18b_defs.h
-rw-r--r--   1 tom  wheel   2080 Mar 20 12:04 pdp18b_diag.txt
[tom@bernice /tank/home/tom/projects/pdp7-unix/build]$ 
</pre>

**Comment by markpizz** [comment]
I had to rename the file attachment to pdp18b_stddev.c.txt.  did you rename the file to remove the .txt?

**Comment by teverett** [comment]
I did.


On Fri, Mar 25, 2016 at 1:24 PM, Mark Pizzolato <notifications@github.com>
wrote:

> I had to rename the file attachment to pdp18b_stddev.c.txt. did you rename
> the file to remove the .txt?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/simh/simh/issues/290#issuecomment-201441386>
>



-- 
A better world shall emerge based on faith and understanding  - Douglas
MacArthur

**Comment by teverett** [comment]
Here's a diff for you, to show that I have the right file

[tom@bernice /tank/home/tom/projects/simh/PDP18B]$ diff -u pdp18b_stddev.c
pdp18b_stddev.c.old

--- pdp18b_stddev.c 2016-03-25 12:41:23.440837620 -0600

+++ pdp18b_stddev.c.old 2016-03-20 12:04:23.485767766 -0600

@@ -409,34 +409,12 @@

     { 0 }

     };



-/* debugging bitmaps */

-#define DBG_REG  0x0001                                 /* trace
read/write registers */

-#define DBG_INT  0x0002                                 /* display
interrupt activities */

-#define DBG_XMT  TMXR_DBG_XMT                           /* display
Transmitted Data */

-#define DBG_RCV  TMXR_DBG_RCV                           /* display
Received Data */

-#define DBG_RET  TMXR_DBG_RET                           /* display Read
Data */

-#define DBG_MDM  TMXR_DBG_MDM                           /* display Modem
Signals */

-#define DBG_CON  TMXR_DBG_CON                           /* display
connection activities */

-#define DBG_TRC  TMXR_DBG_TRC                           /* display trace
routine calls */

-

-DEBTAB tto_debug[] = {

-  {"REG",    DBG_REG, "read/write registers"},

-  {"INT",    DBG_INT, "interrupt activities"},

-  {"XMT",    DBG_XMT, "Transmitted Data"},

-  {"RCV",    DBG_RCV, "Received Data"},

-  {"RET",    DBG_RET, "Read Data"},

-  {"MDM",    DBG_MDM, "Modem Signals"},

-  {"CON",    DBG_CON, "connection activities"},

-  {"TRC",    DBG_TRC, "trace routine calls"},

-  {0}

-};

-

 DEVICE tto_dev = {

     "TTO", &tto_unit, tto_reg, tto_mod,

     1, 10, 31, 1, 8, 8,

     NULL, NULL, &tto_reset,

     NULL, NULL, NULL,

-    &tto_dib, DEV_DEBUG, 0, tto_debug

+    &tto_dib, 0

     };



 /* Clock: IOT routine */

@@ -1130,7 +1108,6 @@



 int32 tto (int32 dev, int32 pulse, int32 dat)

 {

-sim_debug(DBG_TRC, &tto_dev, "tto(dev=%d, pulse=%d, dat=%d)\n", dev,
pulse, dat);

 if (pulse & 001) {                                      /* TSF */

     if (TST_INT (TTO))

         dat = dat | IOT_SKP;

@@ -1151,7 +1128,6 @@

 int32 c;

 t_stat r;



-sim_debug(DBG_TRC, &tto_dev, "tto_svc(buf=0%o)\n", uptr->buf);

 #if defined (KSR28)                                     /* Baudot... */

 if (uptr->buf == BAUDOT_FIGURES)                        /* set figures? */

     tty_shift = 1;

@@ -1159,17 +1135,14 @@

     tty_shift = 0;

 else {

     c = baud_to_asc[uptr->buf | (tty_shift << 5)];      /* translate */

-    sim_debug(DBG_TRC, &tto_dev, "tto_svc(c=0%o)\n", c);



 #else

 c = sim_tt_outcvt (uptr->buf, TT_GET_MODE (uptr->flags) | TTUF_KSR);

-sim_debug(DBG_TRC, &tto_dev, "tto_svc(c=0%o)\n", c);

 if (c >= 0) {



 #endif



     if ((r = sim_putchar_s (c)) != SCPE_OK) {           /* output; error?
*/

-        sim_debug(DBG_TRC, &tto_dev, "tto_svc(r=%d) - reactivating\n", r);

         sim_activate (uptr, uptr->wait);                /* retry? */

         return ((r == SCPE_STALL)? SCPE_OK: r);

         }

@@ -1183,7 +1156,6 @@



 int32 tto_iors (void)

 {

-sim_debug(DBG_TRC, &tto_dev, "tto_iors(INT=%d)\n", TST_INT (TTO));

 return (TST_INT (TTO)? IOS_TTO: 0);

 }



@@ -1191,7 +1163,6 @@



 t_stat tto_reset (DEVICE *dptr)

 {

-sim_debug(DBG_TRC, &tto_dev, "tto_reset()\n");

 tto_unit.buf = 0;                                       /* clear buffer */

 tty_shift = 0;                                          /* clear state */

 CLR_INT (TTO);                                          /* clear flag */

[tom@bernice /tank/home/tom/projects/simh/PDP18B]$

On Fri, Mar 25, 2016 at 1:24 PM, Mark Pizzolato <notifications@github.com>
wrote:

> I had to rename the file attachment to pdp18b_stddev.c.txt. did you rename
> the file to remove the .txt?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/simh/simh/issues/290#issuecomment-201441386>
>



-- 
A better world shall emerge based on faith and understanding  - Douglas
MacArthur

**Comment by markpizz** [comment]
OK.  
1) Change the config line: "set debug stdout" to "set debug some.other.file"
2) Run it again
3) Hit CTRL-E and type quit
4) Look at and provide the contents of some.other.file

**Comment by teverett** [comment]
<pre>
[tom@bernice /tank/home/tom/projects/pdp7-unix/build]$ pdp7 unixv0.simh 

PDP-7 simulator V4.0-0 Beta        git commit id: d2269f8b
CPU	idle disabled
	8192W, EAE
RB: buffering file in memory
Debug output to "some.other.file"
PDP-7 simulator configuration

CPU	idle disabled
CLK	60Hz, devno=00
PTR	devno=01
PTP	devno=02
TTI	devno=03
TTO	devno=04
LPT	disabled
DRM	disabled
RB	devno=71
DT	disabled

Simulation stopped, PC: 00021 (CAL 0)
sim> quit
Goodbye
RB: writing buffer to file
Debug output disabled
[tom@bernice /tank/home/tom/projects/pdp7-unix/build]$ more some.other.file 
Debug output to "some.other.file"
Debug output to "some.other.file" at Fri Mar 25 14:01:53 2016
PDP-7 simulator V4.0-0 Beta        git commit id: d2269f8b
PDP-7 simulator configuration

CPU     idle disabled
CLK     60Hz, devno=00
PTR     devno=01
PTP     devno=02
TTI     devno=03
TTO     devno=04
LPT     disabled
DRM     disabled
RB      devno=71
DT      disabled
DBG(141040981)> TTO TRC: tto_iors(INT=0)

Simulation stopped, PC: 00021 (CAL 0)
sim> quit
Goodbye
RB: writing buffer to file
[tom@bernice /tank/home/tom/projects/pdp7-unix/build]$ 

</pre>

**Comment by markpizz** [comment]
I have no idea why the code running in that simulator on that OS isn't touching the output port, but it really isn't.  The fact number in DBG(141040981) says that 141,040,981 instructions were executed before you hit CTRL-E.  The tto_iors() is called by the simulator on its way back to the sim> prompt.  My above example didn't include that since I hadn't hit CTRL-E yet (I sent what came out of STDOUT).  When I hit CTRL-E I get the same debug line.

I would try to manually move known good binary files (disk and boot) from your OS X environment to the FreeBSD system and use those as input to the simulation test.

**Comment by teverett** [comment]
I honestly don't know either. I can only presume that something is broken
with my OS.

On Fri, Mar 25, 2016 at 2:47 PM, Mark Pizzolato <notifications@github.com>
wrote:

> I have no idea why the code running in that simulator on that OS isn't
> touching the output port, but it really isn't. The fact number in
> DBG(141040981) says that 141,040,981 instructions were executed before you
> hit CTRL-E. The tto_iors() is called by the simulator on its way back to
> the sim> prompt. My above example didn't include that since I hadn't hit
> CTRL-E yet (I sent what came out of STDOUT). When I hit CTRL-E I get the
> same debug line.
>
> I would try to manually move known good binary files (disk and boot) from
> your OS X environment to the FreeBSD system and use those as input to the
> simulation test.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/simh/simh/issues/290#issuecomment-201492660>
>



-- 
A better world shall emerge based on faith and understanding  - Douglas
MacArthur

**Comment by markpizz** [comment]
Can you try my suggestion of moving the binary files from known good sources to the FreeBSD system...

**Comment by teverett** [comment]
for sure.  I'm just heading out, I'll do it today.


On Sun, Mar 27, 2016 at 5:47 AM, Mark Pizzolato <notifications@github.com>
wrote:

> Can you try my suggestion of moving the binary files from known good
> sources to the FreeBSD system...
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/simh/simh/issues/290#issuecomment-202047201>
>



-- 
A better world shall emerge based on faith and understanding  - Douglas
MacArthur

**Comment by teverett** [comment]
Ok, I tried the binary built on my fresh FreeBSD 10.2.  It works fine on the fresh instance but fails on my home box

<pre>
[root@bernice /tank/home/tom/projects/pdp7-unix/build]# ~tom/pdp7 unixv0.simh 

PDP-7 simulator V4.0-0 Beta        git commit id: 72c328e7
CPU	idle disabled
	8192W, EAE
RB: creating new file
RB: buffering file in memory
unixv0.simh-28> set tto debug=TRC
Command not allowed
Debug output to "some.other.file"
PDP-7 simulator configuration

CPU	idle disabled
CLK	60Hz, devno=00
PTR	devno=01
PTP	devno=02
TTI	devno=03
TTO	devno=04
LPT	disabled
DRM	disabled
RB	devno=71
DT	disabled
G2OUT	devno=05
G2IN	devno=43-44

</pre>

**Comment by markpizz** [comment]
When I was referring to the "binary" files I was referring to the binary INPUT files to the simulator (the disk image and boot code), not the simulator executable.  What got tested?

**Comment by teverett** [comment]
I used simh from the other machine.   I can try using the pdp7 files from
the other machine if that would help?


On Sun, Mar 27, 2016 at 12:43 PM, Mark Pizzolato <notifications@github.com>
wrote:

> When I was referring to the "binary" files I was referring to the binary
> INPUT files to the simulator (the disk image and boot code), not the
> simulator executable. What got tested?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/simh/simh/issues/290#issuecomment-202122749>
>



-- 
A better world shall emerge based on faith and understanding  - Douglas
MacArthur

**Comment by teverett** [comment]
I've managed to duplicate the problem in a VirtualBox VM.  Mark could you email me privately at tom@khubla.com please?

**Comment by markpizz** [comment]
I'm repeating the content of my private email to you here to document the result of this poking around before I close the issue.

I just performed the test I had asked you to do:

I copied the a.rim, image.fs and unixv0.simh files (which you originally provided in the binaries directory on the repo) from my FreeBSD 10.2  which displays a “login: “ prompt TO the VirtualBox VM you just made available.

I then used the existing pdp7 binary in your VM to run the unixv0.simh commands and I got a login prompt.

The problem seems to be in the boot.rim file produced by the pdp7-unix project.  If I overwrite the a.rim I’ve got with the boot.rim contents then using the same unixv0.simh script produces the results you are seeing (no Login: prompt).  Actually, if I use the original a.rim as the boot image and the current image.fs it also doesn’t produce a “login: “ prompt.

There is something wrong with the binaries being produced by the pdp7-unix project on this platform.  I don’t have any evidence that there is a simh problem here….

The boot.rim file which the pdp7-unix project is producing (at least on the FreeBSD platform) is only 3 bytes.  This can hardly be useful.
